sklearn/model_selection/tests/test_validation.py

```diff
@@ -396,12 +396,14 @@ def test_cross_validate_return_train_score_warn():
 
     X, y = make_classification(random_state=0)
     estimator = MockClassifier()
-
+    msg_nsplit = ("The default value of n_splits=3 is deprecated in "
+                  "version 0.20 and will be changed to n_splits=5 "
+                  "in version 0.22")
     result = {}
     for val in [False, True, 'warn']:
-        result[val] = assert_no_warnings(cross_validate, estimator, X, y,
-                                         return_train_score=val)
-
+        result[val] = assert_warns_message(FutureWarning, msg_nsplit,
```

Comment:
Firstly this needs to assert that no other warnings are issued. Secondly it needs to be clear to the developer tasked with completing this deprecation in a couple of years' time what it used to be asserting, i.e. that it was asserting an absence of warnings.t

Thirdly, the right solution is actually to pass an explicit value for cv here. Sorry I didn't notice that before.